core/bin/external_node/src/config/mod.rs

```diff
@@ -413,6 +442,229 @@ pub(crate) struct OptionalENConfig {
 }
 
 impl OptionalENConfig {
+    fn from_configs(general_config: &GeneralConfig, enconfig: &ENConfig) -> anyhow::Result<Self> {
+        let api_namespaces = load_config!(general_config.api_config, web3_json_rpc.api_namespaces)
+            .map(|a: Vec<String>| {
+                let result: Result<Vec<_>, _> = a.iter().map(|a| serde_json::from_str(a)).collect();
+                result
+            })
+            .transpose()?;
+
+        Ok(OptionalENConfig {
+            filters_limit: load_optional_config_or_default!(
+                general_config.api_config,
+                web3_json_rpc.filters_limit,
+                default_filters_limit
```

Comment:
I was trying to mimic previous behaviour

Reply:
Do defaults differ between the main node and EN? If not, IMO, it would make sense to refactor it in both places. Even if you don't do that, you can use existing getters in the config.

Reply:
I had one another thought and we have defaults only for EN, just to keep backward compatibility with envs and have them the same if not provided. 

We can make these fields mandatory for file based configuration and that's it

Reply:
> We can make these fields mandatory for file based configuration

This doesn't sound right; these fields should have reasonable defaults. This doesn't mean that types of the corresponding fields should be `Option`al; I'd argue that they shouldn't *because* of the reasonable defaults.

> we have defaults only for EN

That's not true. E.g., for `filters_limit` we have a default value 10_000 exposed via `Web3JsonRpcConfig::filters_limit()` getter. The fact that `Web3JsonRpcConfig.filters_limit` is optional is IMO a flaw (see above).

Reply:
I'd left it for the next iteration, where I'd probably remove these smaller structs in favor of properly compatible ones. 
 
We will have no `OptionalEnConfig` `RequiedEnConfig` and so on , rather general and ExternalNode